packages/amplify_auth_cognito/ios/Classes/FlutterFetchCognitoSessionResult.swift

```diff
@@ -30,70 +31,145 @@ struct FlutterFetchCognitoSessionResult {
         let session = try res.get()
         self.isSignedIn = session.isSignedIn
         self.userSub = try getUserSub(session: session)
+        self.userPoolTokens = try getTokens(session: session)
+        
         self.identityId = try getIdentityId(session: session)
         self.credentials = try getCredentials(session: session)
-        self.userPoolTokens = try getTokens(session: session)
+    
+        
     } catch  {
         throw error as! AuthError
     }
   }
       
   func toJSON() -> Dictionary<String, Any> {
-    
-    return [
-          "isSignedIn": self.isSignedIn,
-          "tokens": self.userPoolTokens,
-          "userSub": self.userSub,
-          "identityId": self.identityId,
-          "credentials": self.credentials        ]
-      }
+    var result: Dictionary<String, Any> = [:]
+    result["isSignedIn"] = self.isSignedIn
+    result["credentials"] = self.credentials
+    result["identityId"] = self.identityId
+    if (!self.userPoolTokens.isEmpty) {
+        result["tokens"] = self.userPoolTokens
+    }
+    if (self.userSub != "") {
+        result["userSub"] = self.userSub
     }
+    return result
+  }}
 
-  func getUserSub(session: AuthSession) throws -> String {
-    var sub: String = ""
+  func getUserSub(session: AuthSession) throws -> String? {
+    let guest = isGuest(session: session)
+    var sub: String? = nil
     if let identityProvider = session as? AuthCognitoIdentityProvider {
       do {
         sub = try identityProvider.getUserSub().get()
       } catch {
-         throw error as! AuthError
+        if (error is AuthError) {
+            if case .signedOut = error as! AuthError {
```

Comment:
Can we simplify to: 

if case .signedOut = error as! AuthError && guest {
return nil
}
else{
throw error as! AuthError
}

Reply:
I don't think this would be valid swift syntax... I don't think "error as! AuthError" actually works as a boolean.